pkg/endpoint/bpf.go

```diff
@@ -465,7 +465,7 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir, reason string) (revnum uint
 	// In the first ever regeneration of the endpoint, the conntrack table
 	// is cleaned from the new endpoint IPs as it is guaranteed that any
 	// pre-existing connections using that IP are now invalid.
-	if !e.ctCleaned {
+	if !e.ctCleaned && !owner.DryModeEnabled() {
```

Comment:
This seems more correct to me, now, but is it directly related to the other changes? I have no idea what dry mode is (seems like it wouldn't be any fun at a party) but if it's unrelated it might be better to have this in a separate commit, in case we broke something.

Reply:
`DryMode` is used for running unit-tests. I did this to clean up annoying errors that were clogging console while running tests locally. I'll factor this out into a separate PR since I want to get rid of `DryModeEnabled` access via `Owner` and grab it from `option.Config` anyway.